### DISCUSSION TITLE
TagoCore v1.0

### DISCUSSION THREAD

**Discussion by felipefdl** [tago-io/tcore#59] [Announcements]
## TagoCore v1.0 is Coming – Here’s What to Expect

We're thrilled to share a big milestone in the TagoCore journey! With the upcoming release of TagoCore v1.0, we’re introducing a fresh, streamlined approach that will make your experience smoother and more powerful than ever. If you’ve been with us during the beta phase, you know TagoCore has always been about adaptability and efficiency. Now, we’re taking that to the next level. 

### The Story Behind the Changes

For a while now, we’ve been listening to feedback and working to create a version of TagoCore that’s not just an update but a transformation. That’s why we’re saying goodbye to the old Plugin Store. In TagoCore v1.0, all plugins will now live together with the main codebase. This shift to a monorepo setup means **public plugins are fully open-source** and right there for you to explore, contribute to, and customize. This not only simplifies things but also aligns with our open and community-driven vision.

As for **Private Plugins**, they’re getting a serious upgrade too. Imagine being able to simply drop your plugin folder into the TagoCore Console without worrying about building or bundling. That’s right—no more compiling hassles, just a clean, direct path from development to deployment. And here’s the cherry on top: both TypeScript and JavaScript plugins work out-of-the-box. Just code, load, and go.

With this evolution, we’re also bidding farewell to the Plugin Store API. This means no more dependence on internet access for deployment. It’s freedom from online constraints, a step toward a more flexible, offline-ready TagoCore.

### Why No More Binaries?

Building and signing binaries was a hefty process that didn’t quite pay off in terms of stability or ease of use. So, we’re focusing our efforts where they matter most—making TagoCore accessible in the following ways:

1. **Docker Images** – This will be the recommended route, providing consistency and ease.
2. **Direct Repository Access** – Clone and run from the repo for a hands-on experience.
3. **NPX Installation** – Coming soon! This is in the pipeline for version v1.5.

### Join Us in Testing!

Your feedback has always been our guiding light. As we approach the final stages of this release, we invite you to test TagoCore v1.0. Keep an eye out for nightly Docker images that will be available soon. Your input will be invaluable in shaping the final touches on this exciting release!

Let’s make TagoCore v1.0 a game-changer, together.

**Comment by douglaszuqueto** [comment]
Good news TagoIO team! I was very happy with the project's return.

I'm already following the development. Count on me!

**Comment by felipefdl** [comment]
Hi @douglaszuqueto 

The Docker beta images are published. You can run them using the tag beta.

```docker run -it tagoio/tagocore:beta```

**Reply by douglaszuqueto** [reply]
Thanks @felipefdl .

I will start the tests 🎯
